CMakeLists.txt

```diff
@@ -3,428 +3,602 @@
+            OS_INTERFACE
+  )
+else()
+  # We don't want this target to only conditionally exist.
+  add_library(vsomeip3-cfg INTERFACE)
+  message(STATUS "Multiple routing managers enabled, \"disabling\" the
+  configuration library.  Note that the unittests depend on this library so you
+  won't likely be able to use the tests in this mode"
+  )
+endif()
+
+# ######################################################################################################################
+# Base library
+# ######################################################################################################################
+file(
+  GLOB
+  vsomeip3_SRC
+  "implementation/endpoints/src/*.cpp"
+  "implementation/logger/src/*.cpp"
+  "implementation/tracing/src/*.cpp"
+  "implementation/message/src/*.cpp"
+  "implementation/plugin/src/*.cpp"
+  "implementation/protocol/src/*.cpp"
+  "implementation/routing/src/*.cpp"
+  "implementation/runtime/src/*.cpp"
+  "implementation/security/src/*.cpp"
+  "implementation/utility/src/*.cpp"
 )
-list(SORT ${VSOMEIP_NAME}-cfg_SRC)
-if (VSOMEIP_ENABLE_MULTIPLE_ROUTING_MANAGERS EQUAL 0)
-    add_library(${VSOMEIP_NAME}-cfg SHARED ${${VSOMEIP_NAME}-cfg_SRC})
-    set_target_properties (${VSOMEIP_NAME}-cfg PROPERTIES VERSION ${VSOMEIP_VERSION} SOVERSION ${VSOMEIP_MAJOR_VERSION})
-    target_compile_features(${VSOMEIP_NAME}-cfg PRIVATE cxx_std_17)
-    if (MSVC)
-        set_target_properties(${VSOMEIP_NAME}-cfg PROPERTIES COMPILE_DEFINITIONS "VSOMEIP_DLL_COMPILATION_PLUGIN")
-    endif()
+if(ENABLE_MULTIPLE_ROUTING_MANAGERS)
+  list(APPEND vsomeip3_SRC "implementation/configuration/src/configuration_impl.cpp")
+endif()
+list(SORT vsomeip3_SRC)
+
+if(WIN32)
+  list(
+    FILTER vsomeip3_SRC
+           EXCLUDE
+           REGEX
+           ".*uds.*"
+  )
+endif()
 
-    target_link_libraries(${VSOMEIP_NAME}-cfg ${VSOMEIP_NAME} ${Boost_LIBRARIES} ${USE_RT} ${DL_LIBRARY} ${SystemD_LIBRARIES})
-endif ()
+add_library(vsomeip3 SHARED)
+target_sources(vsomeip3 PRIVATE ${vsomeip3_SRC})
+target_compile_features(vsomeip3 PRIVATE cxx_std_17)
+target_include_directories(
+  vsomeip3
+  PUBLIC $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/interface>
+         $<BUILD_INTERFACE:${PROJECT_BINARY_DIR}/implementation/configuration/include>
+  INTERFACE $<INSTALL_INTERFACE:${INSTALL_INCLUDE_DIR}>
+)
+if(MSVC)
+  set_target_properties(vsomeip3 PROPERTIES COMPILE_DEFINITIONS "VSOMEIP_DLL_COMPILATION")
+  set_target_properties(
+    vsomeip3
+    PROPERTIES
+    WINDOWS_EXPORT_ALL_SYMBOLS
+    ON
+  )
+endif()
+target_compile_definitions(
+  vsomeip3
+  PRIVATE $<$<CXX_COMPILER_ID:MSVC>:VSOMEIP_DLL_COMPILATION>
+          $<$<BOOL:${ENABLE_MULTIPLE_ROUTING_MANAGERS}>:VSOMEIP_ENABLE_MULTIPLE_ROUTING_MANAGERS>
+          $<$<BOOL:ENABLE_DEFAULT_EVENT_CACHING>:VSOMEIP_ENABLE_DEFAULT_EVENT_CACHING>
+)
+target_include_directories(vsomeip3 SYSTEM PUBLIC $<BUILD_INTERFACE:${Boost_INCLUDE_DIRS}>)
+target_link_libraries(vsomeip3 PRIVATE STACKTRACE_INTERFACE)
```

Comment:
why is this target_link_libraries separate from the previous one?

Reply:
Good call.  Oversight.  Fixing

Reply:
Fixed in 495f00d2